src/main/java/org/broadinstitute/consent/http/resources/DACAutomationRuleResource.java

```diff
@@ -118,15 +118,16 @@ public Response toggleRule(@Auth AuthUser authUser, @PathParam("dacId") Integer
   }
 
   private void validateAdminOrChairForDAC(User user, Integer dacId) {
-    //TODO: harmonize this with the logic in the DACResource so we're consistent.
-    boolean isAdminOrChair = user.hasUserRole(UserRoles.ADMIN) || isChairOfDAC(user, dacId);
+    boolean isAdminOrChair =
+        user.hasUserRole(UserRoles.ADMIN)
+            || user.checkIfUserHasRole(UserRoles.CHAIRPERSON.getRoleName(), dacId);
     if (!isAdminOrChair) {
       throw new ForbiddenException("User does not have access to the specified DAC ID");
     }
   }
 
   private void validateIsChairOfDAC(User user, Integer dacId) {
-    if (!isChairOfDAC(user, dacId)) {
+    if (Boolean.FALSE.equals(user.checkIfUserHasRole(UserRoles.CHAIRPERSON.getRoleName(), dacId))) {
```

Comment:
Not your code, but I wanted to call this out perhaps for a future refactor. The method name, `checkIfUserHasRole`, is somewhat misleading. There are only two roles that have dac ids, so it could be misused to check for admin with a dac id, or SO with a dac id, etc., and never return anything. Perhaps `verifyDACRole`, or something similar.

Reply:
Realized this as well.  Since I'm going to change the method signature, I'm going to rename it now.